src/fides/api/graph/config.py

```diff
@@ -647,6 +653,31 @@ def serialize_erase_after(self, erase_after: Set[CollectionAddress]) -> Set[str]
         },
     )
 
+    @field_validator("partitioning")
+    @classmethod
+    def validate_partitioning(
+        cls, partitioning: Optional[Dict[str, Any]]
+    ) -> Optional[Dict[str, Any]]:
+        if not partitioning:
+            return partitioning
+        if where_clauses := partitioning.get("where_clauses"):
+            if not isinstance(where_clauses, List) or not all(
+                isinstance(where_clause, str) for where_clause in where_clauses
+            ):
+                raise ValueError("`where_clauses` must be a list of strings!")
+            for partition_clause in where_clauses:
+                if matching := match(PARTITION_CLAUSE_PATTERN, partition_clause):
+                    if matching["field_1"] != matching["field_2"]:
```

Comment:
what is field_1 and field_2? can you add an example?

Reply:
what is there is no match then, something didn't fit the pattern and the match was None?

Reply:
are all partitioning clauses expected to be on the same field as well, or do we just care about the pairs?

Reply:
> ~what is field_1 and field_2? can you add an example?~
> 
> EDIT okay got it, parsing your PARTITION_CLAUSE_PATTERN now

let me still clarify things with some comments so this is easier to understand from a quick glance 👍 


> are all partitioning clauses expected to be on the same field as well, or do we just care about the pairs?

yeah, i do think we'd expect that, but i'm not sure that checking for it really helps in preventing malicious input. granted, i don't have a super concrete example of the type of malicious input that the current check prevents, but it does at least put some more bounds on each of the clause strings. i wasn't think that checking across strings provides a whole lot of value - but i guess it could help prevent/clarify user error?

just typing out my thoughts...